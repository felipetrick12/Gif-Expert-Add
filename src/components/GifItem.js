import React from 'react'
import PropTypes from "prop-types";


export default function GifItem({title,url}) {

    return (
        
        <div className='card animate__animated animate__zoomInDown' >
           <p>{title}</p>
           <img src={url} alt={title}/>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  };
