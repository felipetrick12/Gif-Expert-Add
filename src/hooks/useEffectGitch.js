import { useState,useEffect } from "react"
import {GifGridItem} from '../Helpers/GifGridItem'


export const useEffectGitch = (category) => {
    
    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        GifGridItem (category).then(
            imgs=>{
                setTimeout(() => {
                 
                    setstate({
                        data:imgs,
                        loading:false
                    })
                }, 2000);
            }
        )
    }, [category])

    return state;
}
