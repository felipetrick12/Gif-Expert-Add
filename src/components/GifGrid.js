import React from "react";
import { useEffectGitch } from "../hooks/useEffectGitch";
import GifItem from "./GifItem";
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {
   
       const {data:imagen,loading}= useEffectGitch(category);


        return (
            <>

            
            <h2>{category}</h2>
            <p>{loading && 'Cargando'}</p>
            
                <div className='cardGrid '>
                {imagen.map((img) => (
                    <GifItem key={img.id} {...img}/> //se utiliza el expred para enviar propiedades indepenedientes del arreglo
                ))}
               </div>
           
            </>
        );
};
GifGrid.prototype = {
    category: PropTypes.string.isRequired
}
