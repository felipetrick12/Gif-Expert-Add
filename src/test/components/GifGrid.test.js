import { GifGrid } from "../../components/GifGrid"
import {shallow} from 'enzyme'
import { useEffectGitch } from "../../hooks/useEffectGitch"
jest.mock('../../hooks/useEffectGitch')

describe('Prueba de Componente GifGrid', () => {

    const category = 'Hola';
        test('Deberia pasar el componente', () => {
            
            useEffectGitch.mockReturnValue({ //falsear data, para obtener lo que envia el usseefectgitch
                data:[], 
                loading: true
            })
            const wrapper = shallow(<GifGrid category={category} />);
            expect(wrapper).toMatchSnapshot();
        })

        test('Deberia implementar valores a nuestro array imagenes', () => {

            const gifs = [{
                id: 'abc',
                url: 'https://hola.com',
                title: 'hola'
            }]

             useEffectGitch.mockReturnValue({
                 data:gifs,
                 loagind: false
             })

            const wrapper = shallow(<GifGrid category={category} />);

            expect(wrapper).toMatchSnapshot();
            expect(wrapper.find('p').exists()).toEqual(true); //si el parrafo existe deberia ser igual a true
            expect(wrapper.find('GifItem').length).toBe(gifs.length);
            
        })
        
        
})
