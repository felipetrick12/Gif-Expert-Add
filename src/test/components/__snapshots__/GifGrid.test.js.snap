// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de Componente GifGrid Deberia implementar valores a nuestro array imagenes 1`] = `
<Fragment>
  <h2>
    Hola
  </h2>
  <p />
  <div
    className="cardGrid "
  >
    <GifItem
      id="abc"
      key="abc"
      title="hola"
      url="https://hola.com"
    />
  </div>
</Fragment>
`;

exports[`Prueba de Componente GifGrid Deberia pasar el componente 1`] = `
<Fragment>
  <h2>
    Hola
  </h2>
  <p>
    Cargando
  </p>
  <div
    className="cardGrid "
  />
</Fragment>
`;
